File: Backend/src/main/java/org/example/backend/domain/Classroom.java
package org.example.backend.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

@Getter
@Setter
@Entity
@Table(name = "classroom")
public class Classroom {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ColumnDefault("nextval('classroom_id_seq')")
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "type", nullable = false, length = 100)
    private String type;

    @Column(name = "double_size", nullable = false)
    private Boolean doubleSize = false;

}

File: TTSolver/Model/Classroom.py
class Classroom:
    def __init__(self, id: int, name: str, type: str, double_size: bool = None):
        self.id = id
        self.name = name
        self.type = type
        self.double_size = double_size

    def __str__(self):
        return f"Classroom(id={self.id}, name={self.name}, type={self.type}, double_size={self.double_size})"
